lib/concurrent/executor/executor.rb

```diff
@@ -65,22 +65,32 @@ def self.executor(executor_identifier)
       end
     end
 
-    # The policy defining how rejected tasks (tasks received once the
-    # queue size reaches the configured `max_queue`, or after the
-    # executor has shut down) are handled. Must be one of the values
-    # specified in `FALLBACK_POLICIES`.
-    attr_reader :fallback_policy
-
-    # def initialize(opts)
-    #   @auto_terminate = opts.fetch(:auto_terminate, true)
-    # end
-
-    def auto_terminate?
-      mutex.synchronize { ns_auto_terminate? }
+    # @!macro [attach] executor_method_post
+    #
+    #   Submit a task to the executor for asynchronous processing.
+    #
+    #   @param [Array] args zero or more arguments to be passed to the task
+    #
+    #   @yield the asynchronous task to perform
+    #
+    #   @return [Boolean] `true` if the task is queued, `false` if the executor
+    #     is not running
+    #
+    #   @raise [ArgumentError] if no task is given
+    def post(*args, &task)
```

Comment:
@pitr-ch The issue of closures came up a long time ago in a conversation another PR or Issue. Someone had written a (very favorable) blog post in which it was noted that arguments couldn't be passed into one of our abstractions. I can't remember which one (I want to say it was Future, but I honestly can't remember). At issue was relying on the closure vs. a scoped variable. I don't remember the details. The result of the conversation was that I added a feature to allow argument passing. I'll see if I can find the Issue/PR.